### PR TITLE
kindlegen: remove the package (closes #96439)

### LINKED ISSUE #96439: Amazon has killed KindleGen
[Amazon is no longer hosting KindleGen](https://www.amazon.com/gp/feature.html/?docId=1000765211). Consequently, the [nix kindlegen package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/typesetting/kindlegen/default.nix) is broken. I suppose we need to choose whether to remove the package or find a mirror?

CC maintainer: @peterhoeg

Side notes: Sadly, its replacement, [Kindle Previewer](https://www.amazon.com/gp/feature.html?ie=UTF8&docId=1000765261), has no Linux support. But on a more positive note, there are other programs that can produce mobi files, some mentioned in [this discussion](https://news.ycombinator.com/item?id=24275231).

### PR DESCRIPTION
**Pull request by chris-martin**

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5953,8 +5951,6 @@ in
 
   jump = callPackage ../tools/system/jump {};
 
-  kindlegen = callPackage ../tools/typesetting/kindlegen { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:5956]: Can you add kindlegen to `pkgs/top-level/aliases.nix`.

Comment (chris-martin) [reply pkgs/top-level/all-packages.nix:5956]: Done!

**Review by Mic92 — APPROVED** [review]
Can be merged after eval

**Pull request merged**